From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage by mwifiex security fix
Patch-mainline: Never, kABI fix
References: CVE-2019-3846,bsc#1136424

The recent fix for mwifiex driver for a security issue
  patches.fixes/mwifiex-Don-t-abort-on-small-spec-compliant-vendor-I.patch
broke kABI.  Fix by restoring the old firmware struct with ifdef.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/net/wireless/marvell/mwifiex/fw.h |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/drivers/net/wireless/marvell/mwifiex/fw.h
+++ b/drivers/net/wireless/marvell/mwifiex/fw.h
@@ -1743,10 +1743,16 @@ struct mwifiex_ie_types_wmm_queue_status
 struct ieee_types_vendor_header {
 	u8 element_id;
 	u8 len;
+#ifdef __GENKSYMS__
+	u8 oui[4];	/* 0~2: oui, 3: oui_type */
+	u8 oui_subtype;
+	u8 version;
+#else
 	struct {
 		u8 oui[3];
 		u8 oui_type;
 	} __packed oui;
+#endif
 } __packed;
 
 struct ieee_types_wmm_parameter {
@@ -1760,8 +1766,10 @@ struct ieee_types_wmm_parameter {
 	 *   Version     [1]
 	 */
 	struct ieee_types_vendor_header vend_hdr;
+#ifndef __GENKSYMS__
 	u8 oui_subtype;
 	u8 version;
+#endif
 
 	u8 qos_info_bitmap;
 	u8 reserved;
@@ -1780,8 +1788,10 @@ struct ieee_types_wmm_info {
 	 *   Version     [1]
 	 */
 	struct ieee_types_vendor_header vend_hdr;
+#ifndef __GENKSYMS__
 	u8 oui_subtype;
 	u8 version;
+#endif
 
 	u8 qos_info_bitmap;
 } __packed;
